src/librustc/hir/lowering.rs

```diff
@@ -3424,8 +3424,8 @@ pub fn is_range_literal(sess: &Session, expr: &hir::Expr) -> bool {
         ExprKind::Call(ref func, _) => {
             if let ExprKind::Path(QPath::TypeRelative(ref ty, ref segment)) = func.node {
                 if let TyKind::Path(QPath::Resolved(None, ref path)) = ty.node {
-                    let new_call = segment.ident.as_str() == "new";
-                    return is_range_path(&path) && is_lit(sess, &expr.span) && new_call;
+                    let is_new_call = segment.ident.name == sym::new;
+                    return is_range_path(&path) && is_lit(sess, &expr.span) && is_new_call;
```

Comment:
```suggestion
                    return is_new_call && is_range_path(&path) && is_lit(sess, &expr.span);
```
Nit: `is_new_call` is very usually false, so we can skip the further conditions.

Reply:
This change is no longer present.